priv/templates/phx.gen.auth/auth.ex

```diff
@@ -114,13 +142,37 @@ defmodule <%= inspect auth_module %> do
       conn = fetch_cookies(conn, signed: [@remember_me_cookie])
 
       if token = conn.cookies[@remember_me_cookie] do
-        {token, put_token_in_session(conn, token)}
+        {token, conn |> put_token_in_session(token) |> put_session(:<%= schema.singular %>_remember_me, true)}
       else
         {nil, conn}
       end
     end
   end
 
+  # Refresh the session token if it is older than the configured refresh age.
+  defp maybe_refresh_<%= schema.singular %>_session_token(conn, nil), do: conn
+
+  defp maybe_refresh_<%= schema.singular %>_session_token(conn, <%= schema.singular %>_token) do
+    if <%= inspect schema.alias %>Token.seconds_since_refresh(<%= schema.singular %>_token) > @session_refresh_age_in_hours * 60 * 60 do
+      <%= inspect context.alias %>.refresh_<%= schema.singular %>_session_token(<%= schema.singular %>_token.token)
+      maybe_refresh_remember_me_cookie(conn, <%= schema.singular %>_token.token)
```

Comment:
Don't we need to store the new token in the session? As well as in the cookie? And I think the function below can be nuked.